Firestore/core/src/util/byte_stream_apple.mm

```diff
@@ -95,8 +83,23 @@
   return static_cast<int32_t>(read);
 }
 
-StreamReadResult ByteStreamApple::EofResult() {
-  return StreamReadResult(std::move(buffer_), true);
+StreamReadResult ByteStreamApple::ConsumeBuffer() {
+  // NSInputStream does not have consistent behavior with streamStatus, we
+  // perform a "peek" operation here to match with C++ istream implementation,
+  // and make the behavior deterministic.
+  std::string peek_result(1, '\0');
+  auto* data_ptr = reinterpret_cast<uint8_t*>(&peek_result[0]);
+  NSInteger read = [input_ read:data_ptr maxLength:1];
+
+  bool is_eof = (read == 0);
```

Comment:
Optional: the parentheses aren't necessary (but fine to keep if you find they aid readability).

Reply:
Yep, I'd like to keep it.